### PR TITLE
Add additional prolog to thumb mode

### PR DESCRIPTION
**Pull request by HexRoman**

### CODE REVIEW

File: archinfo/arch_arm.py [hunk 1]
```diff
@@ -196,6 +197,7 @@ def is_thumb(self, addr):  # pylint:disable=unused-argument
     }
     thumb_prologs = {
         br"[\x00-\xff]\xb5[\x00-\xff]\xb0",             # push {??, ??, ..., ??, lr}; sub sp, sp, #??
+        br"\xb4\x80\xb0[\x00-\xff]",                    # push {r7}; sub sp, sp, #??
```
Comment (ltfish) [inline archinfo/arch_arm.py]: Are you sure this is little endian THUMB?

**Comment by ltfish** [comment]
I think all the existing `sub sp, sp, #??` should be mapped to `\xb0[\x80-\xff]`. `\xb0[\x00-\x7f]` should be `add sp, sp, #??`.

**Comment by HexRoman** [comment]
> I think all the existing `sub sp, sp, #??` should be mapped to `\xb0[\x80-\xff]`. `\xb0[\x00-\x7f]` should be `add sp, sp, #??`.

I'm not really sure I understood what you meant. My change is adding:
`push {r7}; sub sp, sp, #??` 
Which is a common pattern in embedded arm-thumb binaries for function prologs. 
For example:
```
80 B4       PUSH            {R7}
85 B0       SUB             SP, SP, #0x14
```
Do you think it supposed to look different?

**Comment by ltfish** [comment]
Regarding my comment above: I'm saying all our existing signatures for `sub sp, sp, #??` are incorrect and should be fixed to exclude `add sp, sp, #??`.

**Comment by HexRoman** [comment]
Yes, I see now what you mean, I checked it and you right. 
I updated this in my pull request.

File: archinfo/arch_arm.py [hunk 2]
```diff
@@ -195,9 +196,10 @@ def is_thumb(self, addr):  # pylint:disable=unused-argument
         br"\r\xc0\xa0\xe1\x04\xe0\x2d\xe5",                # mov r12, sp; push {lr}
     }
     thumb_prologs = {
-        br"[\x00-\xff]\xb5[\x00-\xff]\xb0",             # push {??, ??, ..., ??, lr}; sub sp, sp, #??
-        br"[\x00-\xff]\xb4\x00\xb5[\x00-\xff]\xb0",     # push {r?, r?}; push {lr}; sub sp, sp, #??
-        br"[\x00-\xff]\xb0[\x00-\xff]\x90",             # sub sp, sp, #??; str r0, [sp, ?]
+        br"[\x00-\xff]\xb5[\x80-\xff]\xb0",             # push {??, ??, ..., ??, lr}; sub sp, sp, #??
+        br"[\x80-\xff]\xb0\x80\xb4",                    # push {r7}; sub sp, sp, #??
```
Comment (ltfish) [inline archinfo/arch_arm.py]: I believe what you want is `\x80\xb4\xb0[\x80-\xff]` since the first instruction is `push {r7}`.

**Comment by ltfish** [comment]
Merged. Thanks!

**Pull request merged**